### ISSUE TITLE
somaticScore=0 for variants that seem real and were captured by previous superfreq version

### ISSUE THREAD

**Issue by gilhornung** [ChristofferFlensburg/superFreq#27]
Hi Christoffer,

It's me again...

I noticed a lot of somatic variants which were detected by superfreq before the samtools implementation, but now they are not labeled as somatic. When I look at the allVariants Rdata, I see these mutations with  `somaticScore=0`, but for no apparent reason (no flag and seemingly good pbq, pmq and psr). In IGV they also look genuine.

Here is their data:

Individual | Sample | chr | start | end | reference | variant | inGene | severity | effect | f | cov | ref | var | flag | pbq | pmq | psr | somaticScore | germlineLike | dbSNP | dbMAF | dbValidated | ExAC | ExAC_AF | ExAC_Filter | polyPhen | sift | exon | AApos | AAbefore | AAafter | domain | cosmic | isCosmicCensus | cosmicVariantMPM | cosmicGeneMPMPB
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
WS10 | WS10.C | 2 | 225796151 | 225796151 | C | A | DOCK10 | 100 | unknown | 0.9 | 10 | 1 | 9 |   | 0.668394 | 1 | 1 | 0 |   | dbSNP | 0.2001 | TRUE |   | NA |   |   |   |   |   |   |   |   |   | FALSE | 0 | 0 |  
WS10 | WS10.C | 7 | 23530800 | 23530800 | A | G | RPS2P32 | 100 | unknown | 0.414634 | 41 | 24 | 17 |   | 0.336833 | 1 | 0.736385 | 0 |   | dbSNP | 0.07388 | TRUE | ExAC | 0.073 | PASS |   |   |   |   |   |   |   |   | FALSE | 0 | 0 |  
WS10 | WS10.R | 1 | 986475 | 986481 | NNNNNNN | - | AGRN | 100 | unknown | 0.326531 | 49 | 33 | 16 | Sb | 0.246165 | 0.163734 | 0.00025 | 0 |   |   | NA | NA |   | NA |   |   |   |   |   |   |   |   |   | FALSE | 0 | 0 |  
WS10 | WS10.R | 2 | 225796151 | 225796151 | C | A | DOCK10 | 100 | unknown | 0.818182 | 22 | 4 | 18 |   | 0.568053 | 1 | 0.337662 | 0 |   | dbSNP | 0.2001 | TRUE |   | NA |   |   |   |   |   |   |   |   |   | FALSE | 0 | 0 |  
WS10 | WS10.R | 9 | 141070125 | 141070125 | C | T | TUBBP5 | 100 | unknown | 0.916667 | 12 | 1 | 11 |   | 0.280087 | 1 | 1 | 0 |   | dbSNP | 0.3225 | TRUE | ExAC | 0.416 | PASS |   |   |   |   |   |   |   |   | FALSE | 0 | 0 |  

I subsetted the bam files, and you can find them in:
https://owncloud.incpm.weizmann.ac.il/index.php/s/BwM20JfLVsFG76W

Same password as before.

Thanks a lot!

Gil

**Comment by ChristofferFlensburg** [comment]
Hey!

Are you running with a matched normal? These are dbSNPs with a pretty high population frequency, so they wouldn't be called without a matched normal. Maybe you can check the frequency in the normals and the reference normals?

**Comment by gilhornung** [comment]
I am running with a matched Normal, and they do not appear there.

What is the"reference normal"?



On Nov 21, 2017 8:14 AM, "Christoffer Flensburg" <notifications@github.com>
wrote:

Hey!

Are you running with a matched normal? These are dbSNPs with a pretty high
population frequency, so they wouldn't be called without a matched normal.
Maybe you can check the frequency in the normals and the reference normals?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ChristofferFlensburg/superFreq/issues/27#issuecomment-345927817>,
or mute the thread

**Comment by ChristofferFlensburg** [comment]
Can you post the logfile? I don't need the bams for this, as the problem isn't related to reading in the information from the bam.

**Comment by gilhornung** [comment]
The log file is in the link.
https://owncloud.incpm.weizmann.ac.il/index.php/s/BwM20JfLVsFG76W

Just to emphasize - I noticed >300 variants that were in the previous superfreq run (non samtools version) but were discarded in this run. I looked at a few and they all had `somaticScore=0` with no good reason. Here I posted only the ones with the highest frequency (because obviously I trust them more).

**Comment by gilhornung** [comment]
I also made sure, and none of these mutations appear in the normal samples:


chr | start | end | reference | variant | Individual | Sample | inGene | severity | effect | f | cov | ref | var | flag | pbq | pmq | psr | somaticScore | germlineLike | dbSNP | dbMAF | dbValidated | ExAC | ExAC_AF | ExAC_Filter | polyPhen | sift | exon | AApos | AAbefore | AAafter | domain | cosmic | isCosmicCensus | cosmicVariantMPM | cosmicGeneMPMPB
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
2 | 225796151 | 225796151 | C | A | WS10 | WS10.N | DOCK10 | 100 | unknown | 0 | 9 | 9 | 0 |   | 1 | 1 | 1 | 0 |   | dbSNP | 0.2001 | TRUE |   | NA |   |   |   |   |   |   |   |   |   | FALSE | 0 | 0 |  
7 | 23530800 | 23530800 | A | G | WS10 | WS10.N | RPS2P32 | 100 | unknown | 0 | 9 | 9 | 0 |   | 1 | 1 | 1 | 0 |   | dbSNP | 0.07388 | TRUE | ExAC | 0.073 | PASS |   |   |   |   |   |   |   |   | FALSE | 0 | 0 |  
1 | 986475 | 986481 | NNNNNNN | - | WS10 | WS10.N | AGRN | 100 | unknown | 0 | 28 | 28 | 0 |   | 1 | 1 | 1 | 0 |   |   | NA | NA |   | NA |   |   |   |   |   |   |   |   |   | FALSE | 0 | 0 |  
2 | 225796151 | 225796151 | C | A | WS10 | WS10.N | DOCK10 | 100 | unknown | 0 | 9 | 9 | 0 |   | 1 | 1 | 1 | 0 |   | dbSNP | 0.2001 | TRUE |   | NA |   |   |   |   |   |   |   |   |   | FALSE | 0 | 0 |  
9 | 141070125 | 141070125 | C | T | WS10 | WS10.N | TUBBP5 | 100 | unknown | 0 | 7 | 7 | 0 |   | 1 | 1 | 1 | 0 |   | dbSNP | 0.3225 | TRUE | ExAC | 0.416 | PASS |   |   |   |   |   |   |   |   | FALSE | 0 | 0 |

**Comment by ChristofferFlensburg** [comment]
The indel has the "Sb" (strand bias) flag, so that's why that one's not called. The other four all have pretty low coverage in the matched normal, which I think is the reason for the missed call. It's not desired behaviour though, as we want numbers like these to be called. It seems like the algorithm doesn't handle the case of a (decently) high coverage cancer samples and a low coverage matched normal well.

I'll have a closer look and hopefully next patch will address this.

From a QC point of view, I'm still a bit suspicious on the fact that they are all high population frequency dbSNPs though... Are most of the 300 lost variants like that? You may want to check the scatter plot to make sure the matched normal is actually from the same patient.

**Comment by gilhornung** [comment]
Hi,

Some statistics: there are 363 mutations that superfreq "captured" in the previous run but missed in the current run and 484 that are common between the runs. So it is quite a large fraction. of the 363 mutations, 222 are in snpDb, but 141 are not. Here is an image with their frequency distribution:
![image](https://user-images.githubusercontent.com/25452585/33118222-dc0b20d0-cf73-11e7-9370-f05194378b07.png)

I saved the data on all mutations in the owncloud link.

Specifically regarding the indel with the Sb flag, there are 15 reads in one direction and 10 in the other direction, as you can see from the IGV image below. So I think the strand bias was also mis-calculated here, but this is obviously a different error then all the rest of the mutations.
![image](https://user-images.githubusercontent.com/25452585/33118496-b8605d52-cf74-11e7-9921-5f1565017b00.png)

Is there a way to bypass super-freq's mutation calling and add the mutations that were discarded?

Thanks,

Gil

**Comment by ChristofferFlensburg** [comment]
Thanks for the stats. I am increasingly convinced that coverage in the normal is the main issue here. I am looking into fixes.

There is no supported way to bypass the somatic calling. It's a feature that would be nice, but I don't expect to find the time for it anytime soon. The code is decently modular though, with the somatic calling done in markSomatic() in matchFlagVariants.R. External somatic calls would be imported in that function if you want to play with it. In principle it's just a matter of a flag for using external somaticP (the flag would have to be passed from the users superFreq() call), and then import a user supplied .vcf and match it against the variants in superFreq and set the somaticP.

**Comment by gilhornung** [comment]
Thanks.
I am not at my computer now, but I will look at the coverage of the normal
samples in the 300+ mutations.

Where in the code do you think the low normal coverage comes into play?

G

**Comment by ChristofferFlensburg** [comment]
Try 0.9.24 that just went online. Looks good in preliminary tests, and shouldn't have the problem (as much) with low coverage in the matched normal. Let me know how it goes.

**Comment by gilhornung** [comment]
Thanks!
Will try it.

Gil

**Comment by gilhornung** [comment]
Hi Christoffer,

I re-run the patch on my data, but I still get a lot of missed variants (about the same numbers).
What to do?

Gil

**Comment by ChristofferFlensburg** [comment]
Seems like we didn't find the problem, so if you really need them called, I suggest going back to the older version, or try some other variant callers. Maybe try consensus calling with several callers?

**Comment by gilhornung** [comment]
These are a lot of mutations and some have high frequency, so I don't want to lose them.
The problem is not in the variant calling , but rather in the filtering that superFreq does. Is there a way for me to follow what causes a mutation to get somaticP=0 ?

**Comment by ChristofferFlensburg** [comment]
the somatic score is set in markSomatic(). Not the prettiest part of the code, but I dont think any part is, sorry. :P

**Comment by gilhornung** [comment]
When searching for markSomatic I get no results...

**Comment by gilhornung** [comment]
Hi Christoffer,

I looked at a few examples, and it seems that a lot of variants are filtered out because of the following:

1) Low coverage on the normal samples. This leads to high p-values in the binomial test for `pNormalHet`, and then you filter out these positions via `notInNormal`.

2) Low number of variant reads in the tumor. Then the fisher exact test between the Normal and Tumor (`psameF`) does not pass the level of significance you require.

So now I believe it is not a bug, but rather aggressive filtering, which some of it you solved in the current patch (I saw that you changed the condition for `notInNormal`).

So now I feel more assured regarding the results.

Thanks!

Gil
